lib/super_good/solidus_taxjar/transaction_id_generator.rb

```diff
@@ -0,0 +1,45 @@
+module SuperGood
+  module SolidusTaxjar
+    # Responsible for generating `transaction_id` references for transactions
+    # we create on TaxJar for Solidus orders. This class handles creating
+    # associated ID's for transactions which need to be cancelled and recreated
+    # when the order is updated in Solidus after it has been sent to the TaxJar
+    # reporting API.
+    class TransactionIdGenerator
+      class << self
+        # @param order [Spree::Order] the order for which we want to generate
+        #   a `transaction_id`.
+        # @return [String] the initial `transaction_id` for the given order.
+        def initial_transaction_id(order)
+          order.number
+        end
+
+        # Generates the next sequential `transaction_id` from the current one
+        # for the transaction on TaxJar. We do this if the current transaction
+        # has been cancelled and we need to create an updated transaction.
+        #
+        # @param current_transaction_id [String]
+        # @return [String] the next `transaction_id`
+        def next_transaction_id(current_transaction_id)
+          parts = current_transaction_id.split(/-/)
+          if parts.length == 1
```

Comment:
If the order number has a dash in it, this falls apart.

Could we do if `current_transaction_id == order.number` instead to deal with the base case?

Reply:
Great catch, we considered that for the second case but not for this initial one. I like your recommendation, we'll add a spec for this and update the change to handle that case!